kde/frameworks/extra-cmake-modules/icotool.diff

```diff
@@ -0,0 +1,153 @@
++#      recommended because png2ico does not support vista icons with 256px.
++#      See :find-module:`FindPng2Ico` or :find-module:`FindIcotool`.
++#    * Supported sizes: 16, 32, 48, 64, 128. icotool also supports 256px.
+ #
+ # Mac OS X notes
+ #    * The executable target must have the ``MACOSX_BUNDLE`` property set.
+@@ -161,11 +163,15 @@ function(ecm_add_app_icon appsources)
+         message(AUTHOR_WARNING "No 128px icon provided; this will not work on Mac OS X")
+     endif()
+ 
++    # 16x16, 32x32, 48x48, and 256x256 (code scales between 32 and 256)
++    # For Classic Mode, the full set is 16x16, 24x24, 32x32, 48x48 and 64x64.
+     set(windows_icons ${icons_at_16px}
++                      ${icons_at_24px}
+                       ${icons_at_32px}
+                       ${icons_at_48px}
+                       ${icons_at_64px}
+-                      ${icons_at_128px})
++                      ${icons_at_128px}
++                      ${icons_at_256px})
+     if (NOT windows_icons)
+         message(AUTHOR_WARNING "No icons suitable for use on Windows provided")
+     endif()
+@@ -176,25 +182,14 @@ function(ecm_add_app_icon appsources)
+         set(saved_CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}")
+         set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_FIND_MODULE_DIR})
+         find_package(Png2Ico)
++        find_package(IcoTool)
+         set(CMAKE_MODULE_PATH "${saved_CMAKE_MODULE_PATH}")
+ 
+-        if (Png2Ico_FOUND)
+-            if (Png2Ico_HAS_RCFILE_ARGUMENT)
+-                add_custom_command(
+-                    OUTPUT "${_outfilename}.rc" "${_outfilename}.ico"
+-                    COMMAND Png2Ico::Png2Ico
+-                    ARGS
+-                        --rcfile "${_outfilename}.rc"
+-                        "${_outfilename}.ico"
+-                        ${windows_icons}
+-                    DEPENDS ${windows_icons}
+-                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+-                )
+-            else()
++        function(create_windows_icon_and_rc command args)
+                 add_custom_command(
+                     OUTPUT "${_outfilename}.ico"
+-                    COMMAND Png2Ico::Png2Ico
+-                    ARGS "${_outfilename}.ico" ${windows_icons}
++                    COMMAND ${command}
++                    ARGS -c -o "${_outfilename}.ico" ${windows_icons}
+                     DEPENDS ${windows_icons}
+                     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+                 )
+@@ -207,10 +202,53 @@ function(ecm_add_app_icon appsources)
+                     DEPENDS "${_outfilename}.ico"
+                     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+                 )
+-            endif()
++        endfunction()
++
++        if (IcoTool_FOUND)
++            set(icotool_args "-c -o \"${_outfilename}.ico\"")
++
++            foreach(size 16 32 48)
++                foreach(icon ${icons_at_${size}px})
```

Comment:
what about 48 < x < 256

Reply:
They are not part of the "classic full set".

https://msdn.microsoft.com/en-us/library/windows/desktop/dn742485%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

```
Application icons and Control Panel items: The full set includes 16x16, 32x32, 48x48, and 256x256 (code scales between 32 and 256). The .ico file format is required. For Classic Mode, the full set is 16x16, 24x24, 32x32, 48x48 and 64x64.```

As I understand it, these are the only sizes that are supported for *application* icons. For other usages this macro is not made and would need to be adjusted anyway.

Reply:
hm sounds good